### PR TITLE
[binder] Truncate for aggregate connection, fix inner_sym

### PR DESCRIPTION
**Pull request by SpriteOvO**
Some fixes for binder:

- Truncate for aggregate connection.

- Set `inner_sym` only when needed.

### CODE REVIEW

File: binder/src/main/scala/PanamaCIRCTConverter.scala [hunk 1]
```diff
@@ -574,13 +581,136 @@ class PanamaCIRCTConverter extends CIRCTConverter {
...
-        .withNamedAttr("inner_sym", circt.hwInnerSymAttrGet(name))
         .withOperand(input)
         .withResult(util.convert(resultType))
         // .withResult( /* ref */ )
         .build()
+      firCtx.ops += ((id._id, op.op))
       firCtx.newItem(id, op.results(0))
     }
+
+    def emitConnect(dest: Reference.Value, srcVal: Reference.Value, loc: MlirLocation): Unit = {
```
Comment (sequencer) [inline binder/src/main/scala/PanamaCIRCTConverter.scala:592]: This function is too mutable, before real releasing, we need to refactor.

File: binder/src/main/scala/PanamaCIRCTConverter.scala [hunk 2]
```diff
@@ -574,13 +581,136 @@ class PanamaCIRCTConverter extends CIRCTConverter {
...
+      val indexType = circt.mlirIntegerTypeGet(32)
+      var src = srcVal
+
+      // TODO: Strict connect
+
+      if (dest.tpe.isInstanceOf[fir.BundleType]) {
```
Comment (sequencer) [inline binder/src/main/scala/PanamaCIRCTConverter.scala:598]: Use pattern match.

File: binder/src/main/scala/PanamaCIRCTConverter.scala [hunk 3]
```diff
@@ -574,13 +581,136 @@ class PanamaCIRCTConverter extends CIRCTConverter {
...
+          if (destBundle.fields(index).flip == fir.Flip) {
+            emitConnect(srcField, destField, loc)
+          } else {
+            emitConnect(destField, srcField, loc)
+          }
+        }
+        return
+      }
+
+      if (dest.tpe.isInstanceOf[fir.VectorType]) {
```
Comment (sequencer) [inline binder/src/main/scala/PanamaCIRCTConverter.scala:633]: use pattern match

File: binder/src/main/scala/PanamaCIRCTConverter.scala [hunk 4]
```diff
@@ -574,13 +581,136 @@ class PanamaCIRCTConverter extends CIRCTConverter {
...
+              .withNamedAttrs(Seq(("amount", circt.mlirIntegerAttrGet(circt.mlirIntegerTypeGet(32), destWidth))))
+              .withOperands(Seq(src.value))
+              .withResult(util.convert(dest.tpe))
+              .build()
+              .results(0),
+            dest.tpe
+          )
+        }
+      } else {
+        // TODO: const cast
```
Comment (sequencer) [inline binder/src/main/scala/PanamaCIRCTConverter.scala:705]: match and throw unimplemented error

File: binder/src/main/scala/PanamaCIRCTConverter.scala [hunk 5]
```diff
@@ -574,13 +581,136 @@ class PanamaCIRCTConverter extends CIRCTConverter {
...
         .build()
+      firCtx.ops += ((id._id, op.op))
       firCtx.newItem(id, op.results(0))
     }
+
+    def emitConnect(dest: Reference.Value, srcVal: Reference.Value, loc: MlirLocation): Unit = {
+      val indexType = circt.mlirIntegerTypeGet(32)
+      var src = srcVal
+
+      // TODO: Strict connect
```
Comment (sequencer) [inline binder/src/main/scala/PanamaCIRCTConverter.scala:596]: match and throw unimplemented error

**Review by sequencer — CHANGES_REQUESTED** [review]
Nitpick on the C++ style of writing Scala

File: binder/src/main/scala/PanamaCIRCTConverter.scala (hunk 5 above)
Comment (SpriteOvO) [reply binder/src/main/scala/PanamaCIRCTConverter.scala:596]: The determination of strict connection may need adding something to FIRRTL C-API on the CIRCT side, I prefer just leave a comment here at the moment and fix it later.

File: binder/src/main/scala/PanamaCIRCTConverter.scala (hunk 4 above)
Comment (SpriteOvO) [reply binder/src/main/scala/PanamaCIRCTConverter.scala:700]: The same reason as strict connection.

**Review by sequencer — APPROVED** [review]

**Pull request merged**